### PR TITLE
Provide hook for executing additional handlers

### PR DESCRIPTION
**Pull request by rhyshort**
## What
Provide a hook into `CouchOperation` to make it possible to call additional completion handlers without reimplementing response processing logic.

## How
Added a new hook into CouchOperation processResponse(json: [String:AnyObkect]) to enable
running of additional completion handlers.

## Issues
Part of #43

### CODE REVIEW

File: Source/CouchOperation.swift [hunk 1]
```diff
@@ -147,7 +147,7 @@ public class CouchOperation : NSOperation, HTTPRequestOperation
         do {
             if let data = data {
                 let json = try NSJSONSerialization.jsonObject(with: data) as! [String:AnyObject]
-                
+                self.processResponse(json: json)
```
Comment (brynh) [inline Source/CouchOperation.swift]: This should be in the `if httpInfo.statusCode / 100 == 2` branch.

File: Source/CouchOperation.swift [hunk 2]
```diff
@@ -167,6 +167,19 @@ public class CouchOperation : NSOperation, HTTPRequestOperation
         }
         
     }
+    
+    /**
+     This method is called from the
+     `processResponse(data: NSData?, httpInfo: HttpInfo?, error: ErrorProtocol?)` method,
+     it will contain the desearlized json response in the event the request returned with a
```
Comment (brynh) [inline Source/CouchOperation.swift]: Typo "deserialized"

File: Source/CouchOperation.swift [hunk 3]
```diff
@@ -167,6 +167,19 @@ public class CouchOperation : NSOperation, HTTPRequestOperation
...
+     2xx status code. 
+     
+     - Note: This should be used to trigger other handlers such as a handler for
+     each row of view.
```
Comment (brynh) [inline Source/CouchOperation.swift]: I think this should mention the fact it needs to be overridden - how about `- Note: This should be overridden to trigger other handlers such as a handler for each row of a returned view.`?

File: Source/CouchOperation.swift (hunk 2 above)
Comment (rhyshort) [inline Source/CouchOperation.swift]: done in 54256c0

File: Source/CouchOperation.swift (hunk 3 above)
Comment (rhyshort) [inline Source/CouchOperation.swift]: done in 54256c0

File: Source/CouchOperation.swift (hunk 1 above)
Comment (rhyshort) [inline Source/CouchOperation.swift]: fixed in 4c0c0c1

**Comment by brynh** [comment]
+1

**Pull request merged**